modules/p2/ding_entity_rating/ding_entity_rating.module

```diff
@@ -57,7 +57,17 @@ function ding_entity_rating_get_user_rating_callback() {
     $ratings[$object->ding_entity_id] = ding_entity_rating_get_user_rating($object);
   }
 
-  echo drupal_json_encode($ratings);
+  $missing = array_keys(array_filter($ratings, function ($v) {
+    return $v === FALSE;
+  }));
+
+  $missing = ding_provider_invoke('openlist', 'call_module', 'TingObjectRating', 'getRatings', array(
+    $missing,
+  ));
+
+  $ratings = array_filter($ratings);
+
+  echo drupal_json_encode(array('user_ratings' => $ratings, 'averages' => $missing));
```

Comment:
Variable naming and handling makes this code hard to follow. As I see it:

L58: `$ratings` now contain a map from object ids to the users' own rating. Values will be `FALSE` if the user has not rated the object.

L64: `$missing` now contain an array of object ids which the user has not rated yet.

L66: `$missing` now contain a map from object ids to global average rating.

L68: `$ratings` now contain a map from object ids which the user has actually to rated to the user rating. 

At least use a new more appropriately named variable on L64.